src/lib.rs

```diff
@@ -599,6 +599,68 @@ impl OutputPolicy {
     }
 }
 
+/// Structure that proofs that certain user agrees with
+/// some data. This structure is used to sign `Transaction`
+/// and get `SignedTransaction` out.
+///
+/// It's important that witness works with opaque structures
+/// and may not know the contents of the internal transaction.
+#[wasm_bindgen]
+#[derive(Clone)]
+pub struct Witness(tx::Witness);
+
+impl From<tx::Witness> for Witness {
+    fn from(witness: tx::Witness) -> Witness {
+        Witness(witness)
+    }
+}
+
+#[wasm_bindgen]
+impl Witness {
+    /// Generate Witness for an utxo-based transaction Input
+    pub fn for_utxo(
+        genesis_hash: Hash,
+        transaction_id: TransactionSignDataHash,
+        secret_key: PrivateKey,
+    ) -> Witness {
+        Witness(tx::Witness::new_utxo(
+            &genesis_hash.0,
+            &transaction_id.0,
+            &secret_key.0,
+        ))
+    }
+
+    /// Generate Witness for an account based transaction Input
+    /// the account-spending-counter should be incremented on each transaction from this account
+    pub fn for_account(
+        genesis_hash: Hash,
+        transaction_id: TransactionSignDataHash,
+        secret_key: PrivateKey,
+        account_spending_counter: SpendingCounter,
+    ) -> Witness {
+        Witness(tx::Witness::new_account(
+            &genesis_hash.0,
+            &transaction_id.0,
+            &account_spending_counter.0,
+            &secret_key.0,
+        ))
+    }
+
+    /// Get string representation
+    pub fn to_bech32(&self) -> Result<String, JsValue> {
+        let bytes = self
+            .0
+            .serialize_as_vec()
+            .map_err(|error| JsValue::from_str(&format!("{}", error)))?;
+
+        Bech32::new("witness".to_string(), bytes.to_base32())
+            .map(|bech32| bech32.to_string())
+            .map_err(|error| JsValue::from_str(&format!("{}", error)))
+    }
+}
+
+impl_collection!(Witnesses, Witness);
+
```

Comment:
is there a reason to have moved all of these from their original place in the file?

Reply:
Ah sorry this is a leftover from me messing with macros. I moved it back